packages/calypso-e2e/src/lib/flows/difm-flow.ts

```diff
@@ -0,0 +1,89 @@
+import { Page } from 'playwright';
+import { DataHelper } from '../..';
+
+const selectors = {
+	// General
+	button: ( text: string ) => `button:text("${ text }")`,
+
+	// Start page
+	startContainer: '.signup.is-do-it-for-me',
+
+	// Options page
+	siteName: '#siteTitle',
+	tagline: '#tagline',
+
+	// Social page
+	facebook: 'input[name=facebookUrl]',
+};
+
+/**
+ * Class encapsulating the flow when starting a new do-it-for-me site order.
+ */
+export class DIFMFlow {
+	/**
+	 * Constructs an instance of the flow.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( private page: Page ) {}
+
+	/**
+	 * Given text, click on the button's first instance with the text.
+	 *
+	 * @param {string} text User-visible text on the button.
+	 */
+	async clickButton( text: string ): Promise< void > {
+		const selector = selectors.button( text );
+
+		await this.page.waitForSelector( selector );
```

Comment:
Ah did not know that learning something new every day 😄 I actually took this method from another instance. Would be a nice follow-up to clean this up on every occurrence.

This might actually be a good candidate to add to a helper and have the implementation in one place due to its reusability. 🤔

Reply:
>Might actually be a good candidate for another PR to add this to a helper and have the implementation in one place due to its reusability

That's a great question, and actually, one that Edwin and I have waffled back and forth on a lot!

While the DRYness of abstracting out a shared button clicking function is appealing, we've opted for now to just let each POM class decide whether it makes sense to expose a method like this or not.

The main reason for that is there may be situations where the underlying selector for a button (by text) may actually be more complicated than it is in this case. For example, we have a few places throughout Calypso where there are some nested iFrames. These would require extra handling for that frame. There are also might be places where we might have to make the selector a little more specific with a parental selector first. Because that underlying implementation might change, we figured it might be more confusing if there were some places that used a shared library method, but then other places that used POM-specific methods. This way, while more verbose, keeps a consistent pattern of deferring button handling to each POM class. 

Let me know if that doesn't make sense! 👍

Reply:
Makes perfect sense. Did no know about the different implementations 👍🏻